dataloader/dataloader.hpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * @file dataloader.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Dataloader for popular datasets.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core/data/scaler_methods/min_max_scaler.hpp>
+#include <mlpack/core/math/shuffle_data.hpp>
+#include <mlpack/core/data/split_data.hpp>
+#include <mlpack/prereqs.hpp>
+#include <dataloader/datasets.hpp>
+#include <mlpack/core.hpp>
+#include <utils/utils.hpp>
+
+
+#ifndef MODELS_DATALOADER_HPP
+#define MODELS_DATALOADER_HPP
+
+template<
+  typename DataSetX = arma::mat,
```

Comment:
Might be easier to type as `DatasetX` (and lowercased `dataset` matches mlpack documentation's usage a bit better).  Your call. :+1: